src/main/java/appeng/core/sync/packets/PacketCraftRequest.java

```diff
@@ -38,79 +39,71 @@
 import appeng.container.implementations.ContainerCraftConfirm;
 import appeng.core.AELog;
 import appeng.core.sync.AppEngPacket;
+import appeng.core.sync.AppEngPacketHandler;
 import appeng.core.sync.GuiBridge;
-import appeng.core.sync.network.INetworkInfo;
 import appeng.util.Platform;
 
 
-public class PacketCraftRequest extends AppEngPacket
+public class PacketCraftRequest implements AppEngPacket, AppEngPacketHandler<PacketCraftRequest, AppEngPacket>
 {
 
-	public final long amount;
-	public final boolean heldShift;
+	private long amount;
+	private boolean heldShift;
 
 	// automatic.
-	public PacketCraftRequest( ByteBuf stream )
+	public PacketCraftRequest()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)